### PR TITLE
Pull in key changes from AC2 update.

### PR DESCRIPTION
**Pull request by marciaga**
* Asana ticket [link](https://app.asana.com/0/1201700438643002/1203898367643368/f)
* Update README instructions for Waves 02-06 including:

>  * Add `id` instance variable to `Item` class and set default via `uuid` library
>  * Don't use `category` instance variable in `Item` class
>  * Use `category` class variable, set to class's name, e.g. "Item"
>  * Add `get_by_id` instance method to `Vendor` class (returns an `Item` instance from its `inventory` list)
>  * Add `get_category` instance method to `Item` class that returns the `category` class variable
 > * When adding the Electronics, Clothing, and Decor classes, add id instance variable to each. We can then use and override the Item classes __str__ to generate slightly different strings for each class. Same goes for using the Item class's get_category method.

* Small grammatical/tone changes to README (borrowed from [AC2 PR](https://github.com/AdaGold/swap-meet/pull/30))

* Update/modify/add additional tests according to new instructions. 

* See also [Slack thread](https://adadevelopersacademy.slack.com/archives/C037515RYVB/p1676587048612229) with other instructors' input

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,42 +77,50 @@ In Wave 1 we will create the `Vendor` class.
...
 
 ### Wave 3
 
-In Wave 3 we will write a method to stringify an `Item` using `str()` and write the method `swap_items`.
+In Wave 3 we will write a method to stringify (convert to a string) an `Item` using `str()` and write the method `swap_items`.
 
-- When we stringify (convert to a string) an instance of `Item` using `str()`, it returns `"Hello World!"`
+- When we stringify an instance of `Item` using `str()`, it returns `"An object of type Item with id <id value>"`, where `<id value>` is the `id` of the `Item` instance that `str()` was called on.
+  - For example, if we had an `Item` instance `item_a = Item(id=12345)`, the output of `str(item_a)` should be `"An object of type Item with id 12345"`.
   - This implies `Item` overrides its stringify method. We may need to research the `__str__` method for more details!
```
Comment (kyra-at-ada) [inline README.md:103]: I'm wondering if instead of asking folks to research the `__str__` method, we instead add a section at the bottom of the README with a brief explanation of operator overloading and what we might expect `str(item_instance)` to return if we didn't write our own `str()` method. Or we could even include the explanation directly in the bullet points here.

I see a lot of folks in internship who don't seem to fully understand the concept.

File: README.md [hunk 2]
```diff
@@ -79,42 +77,50 @@ In Wave 1 we will create the `Vendor` class.
...
-  - It takes one argument: a string, representing a category
-  - This method returns a list of `Item`s in the inventory with that category
+- Each `Item` will have an attribute named `id`, which is a unique integer by default
+  - There are many ways to generate numbers, but generating numbers without duplicates takes some care. Happily, Python has a package called `uuid` that can help!
+    - If we import the [`uuid` package](https://docs.python.org/3/library/uuid.html) in `item.py`, with a little research we can use one of the functions `uuid` provides to create large unique numbers meant to be used as identifiers
+    - This package creates `UUID` objects, its functions don't directly return an integer, **but** `UUID` instances have [an attribute `int`](https://docs.python.org/3/library/uuid.html#uuid.UUID.int) which allow us to access their value as an integer
+- When we initialize an instance of `Item`, we can optionally pass in an integer with the keyword argument `id` to manually set the `Item`'s `id`
+- Each `Item` will have a function named `get_category`, which will return a string holding the name of the class
```
Comment (kyra-at-ada) [inline README.md:90]: Would it be too much of a give away to add a bullet point that gives an example of how we can create an integer UUID?

ex) `example_uuid = uuid.uuid4().int`
or `example_uuid = uuid.int`

We would probably also want to explain why we use uuid4() or perhaps give both options and encourage them to research why the former may be the better option.

**Review by kyra-at-ada — APPROVED** [review]
These changes look great Mark! 

I made sure I could alter my previous solution to pass all the tests and checked that the tests provide good coverage. 

A couple of my comments aren't actually related to these changes, but things I noticed reading through the project.

 I can't comment on it because there weren't any changes, but the first sentence in the Project Directions (line 53) stood out to me as something we should consider changing given our discussions about students being over reliant on tests. We may want to consider adding a line here to encourage students to still think about whether the provided tests miss any edge cases and ensuring their solution aligns with the project directions in addition to passing any tests.

**Review by kyra-at-ada — APPROVED** [review]
LGTM! Thanks for updating!
